lib/tag-suggestions/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import appState from '../flux/app-state';
+import { tracks } from '../analytics';
+
+const { search } = appState.actionCreators;
+const { recordEvent } = tracks;
+
+export class TagSuggestions extends Component {
+  static displayName = 'TagSuggestions';
+
+  static propTypes = {
+    query: PropTypes.string.isRequired,
+    filteredTags: PropTypes.array.isRequired,
+  };
+
+  render() {
+    const { filteredTags } = this.props;
+
+    return (
+      <Fragment>
+        {filteredTags.length > 0 && (
+          <div className="tag-suggestions">
+            <div className="note-list-header">Tags</div>
+            <ul className="tag-suggestions-list">
+              {filteredTags.map(tag => (
+                <li
+                  key={tag.id}
+                  id={tag.id}
+                  className="tag-suggestion-row"
+                  onClick={() => this.props.onSearch(`tag:${tag.id}`)}
+                >
+                  <div
+                    className="tag-suggestion"
+                    title={decodeURIComponent(tag.id)}
+                  >
+                    tag:{decodeURIComponent(tag.id)}
+                  </div>
+                </li>
+              ))}
+            </ul>
+          </div>
+        )}
+      </Fragment>
+    );
+  }
+}
+
+const mapStateToProps = ({ appState: state }, ownProps) => ({
+  filteredTags: state.tags
+    .filter(function(tag) {
+      // todo split on spaces to support additive query args
+
+      // prefix tag ID with "tag:" (encoded) this allows us to match if the user typed the prefix
+      // n.b. doing it in this direction instead of stripping off any "tag:" prefix allows support
+      // of tags that contain the string "tag:" ¯\_(ツ)_/¯
+      var testID = 'tag%3A' + tag.id;
```

Comment:
note that there is `tag.id` and there should also be `tag.name`. `tag.name` is probably what we want - probably what renders when viewing tags.

Reply:
`tag.data.name` good catch

Reply:
(though they appear to be identical in all cases)

Reply:
actually, that's a lie, the ID is a URI encoded version of the name?
<img width="224" alt="Screen Shot 2019-11-14 at 9 25 03 PM" src="https://user-images.githubusercontent.com/52152/68919012-4588d080-0725-11ea-9c63-cab529f14a4b.png">

Reply:
there is some old confusion on tag `id` vs. `name` and the gist of it is that the `id` is the account-unique id value and is meaningless. the `name` is what displays and can actually change.

there were times where one app used `id` and another `name` and then we had duplicate tags and we lost tags during tag-rename operations.